lib/cog/models/group.ex

```diff
@@ -28,12 +29,30 @@ defmodule Cog.Models.Group do
   detail_fields [:id, :name]
 
   @doc """
-  Creates a changeset based on the `model` and `params`.
-
   If no params are provided, an invalid changeset is returned
   with no validation performed.
   """
-  def changeset(model, params \\ :empty) do
+  def changeset(model), do: changeset(model, :empty)
+
+  @doc """
+  Creates a changeset based on the `model` to validate a delete
+  action.
+  """
+  def changeset(model, :delete) do
+    changeset = %{Changeset.change(model) | action: :delete}
+
+    if model.name == Cog.admin_group do
+      changeset
+      |> add_error(:name, "admin group may not be modified")
```

Comment:
Right, that's exactly what I attempted to do with the `changeset(model, :delete)` functions, so that we'd bail in Elixir before we hit the Postgrex exception. I also couldn't find a clean way to turn those into something useful to a user, unfortunately. For this PR, I'm going to update the changesets to bail when attempting to edit the name of the cog-admin role or group and call this one done.